docs/architecture.md

```diff
@@ -0,0 +1,222 @@
+);
+
+CREATE INDEX piece_content_idx ON piece (content);
+CREATE INDEX piece_inserted_idx ON piece (inserted);
+
+-- Content for which we need to derive piece CIDs. We will have a process that
+-- reads from this queue and writes into `piece` table.
+CREATE VIEW content_queue AS
+  SELECT content.*
+  FROM content
+  LEFT OUTER JOIN piece ON content.link = piece.content
+  WHERE piece.content IS NULL
+  ORDER BY piece.inserted;
+
+-- Table describing pieces to be included into aggregates. If aggregate is NULL then the
+-- piece is queued for the aggregation.
+CREATE TABLE inclusion
+(
+  -- Piece CID. Notice that it is not unique because in case of bad piece
+  -- aggregate will be rejected and good pieces will be written back here
+  -- to be included into new aggregate
+  piece TEXT NOT NULL REFERENCES piece(link),
+  
+  -- Aggregate CID, if NULL the the piece is queued for the aggregation
+  aggregate TEXT REFERENCES aggregate(link) NULL,
+
+  -- Priority in the queue. I think it could be a counter on initial inserts hence
+  --  providing FIFO order. When piece is retried (after aggregate is rejected)
+  -- we could keep original priority hence prioritizing it over new items yet
+  -- keeping original FIFO order among rejects.
+  priority TEXT NOT NULL,
+
+  -- Time when the piece was added to the queue.
+  inserted TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+
+  -- Piece may end up in multiple aggregates e.g. if aggregate was rejected
+  PRIMARY KEY (aggregate, piece)
+
+   -- We may also want to write inclusion proof here
+);
+
+CREATE INDEX inclusion_inserted_idx ON inclusion (inserted);
+
+-- Table for created aggregates. 
+CREATE TABLE aggregate
+(
+  -- commP of the aggregate (commP of the commPs)
+  link TEXT PRIMARY KEY NOT NULL,
+  -- Aggregate size
+  size number NOT NULL,
+  -- Time when the aggregate was created
+  inserted TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL
+);
+
+-- View for inclusion records that do not have an aggregate.
+CREATE VIEW cargo AS
+  SELECT *
+  FROM inclusion
+  WHERE aggregate IS NULL
+  ORDER BY inserted;
+
+-- State of aggregate deals. When aggregate is sent to spade-proxy status is 'PENDING'.
+-- When spade-proxy requests a wallet signature, status will be updated to 'SIGNED'. Once
+-- deal is accepted status changes to `APPROVED`, but if deal fails status will be set to
+-- `REJECTED`.
+CREATE TABLE deal (
+  aggregate TEXT PRIMARY KEY NOT NULL REFERENCES aggregate(link),
+  status DEAL_STATUS DEFAULT 'PENDING',
+  -- if status is an error this may contain details about the error e.g. json containing
+  -- piece CIDs that were invalid.
+  detail TEXT,
+  -- Time when aggregate was send to spade-proxy
+  inserted TIMESTAMP WITH TIME ZONE DEFAULT timezone('utc'::text, now()) NOT NULL,
+  -- Time when aggregate was signed
+  signed TIMESTAMP WITH TIME ZONE,
+  -- Time when aggregate was processed
+  processed TIMESTAMP WITH TIME ZONE
```

Comment:
Does processed here implies when it became either `APPROVED` or `REJECTED` ? It may be good idea to add comment to that effect to make it clear.